src/libs/actions/IOU.js

```diff
@@ -3723,6 +3715,26 @@ function getIOUReportID(iou, route) {
     return lodashGet(route, 'params.reportID') || lodashGet(iou, 'participants.0.reportID', '');
 }
 
+/**
+ * @param {String} transactionID
+ * @param {String} comment
+ * @param {Boolean} isDraft
+ * @param {String} iouType
+ * @param {String} reportID
+ */
+function setMoneyRequestDescription_temporaryForRefactor(transactionID, comment, isDraft, iouType, reportID) {
+    if (isDraft) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {comment: {comment}});
+    } else {
+        if (iouType === CONST.IOU.TYPE.REQUEST) {
+            updateMoneyRequestDescription(transactionID, reportID, comment);
+        }
+        if (iouType === CONST.IOU.TYPE.SPLIT) {
+            setDraftSplitTransaction(transactionID, {comment});
+        }
+    }
+}
```

Comment:
I don't think the migration is being done correctly. Here we should only update the transaction or transaction draft based on the isDraft param.

updateMoneyRequestDescription should be called after setMoneyRequestDescription (not inside)

setDraftSplitTransaction call can be avoided, just make this function handle its logic i.e. use a different collection for draft


Please use the two linked migration examples for reference.